tests/tests_change_settings.yml

```diff
@@ -0,0 +1,78 @@
+- hosts: all
+  tasks:
+    - name: Ensure required packages are installed
+      package:
+        name: tuned
+        state: present
+
+    - name: Ensure kernel settings profile directory exists
+      file:
+        path: /etc/tuned/kernel_settings
+        state: directory
+        mode: 0755
+
+    - name: Generate a configuration for kernel settings
+      copy:
+        src: tuned/etc/tuned/basic_settings/tuned.conf
+        dest: /etc/tuned/kernel_settings/tuned.conf
+        mode: 0644
+
+    - name: Ensure required services are enabled and started
+      service:
+        name: tuned
+        state: restarted
+        enabled: yes
+
+    - name: apply kernel_settings
+      include_role:
+        name: linux-system-roles.kernel_settings
+      vars:
+        kernel_settings_sysctl:
+          - name: fs.file-max
+            value: 400000
+          - name: kernel.threads-max
+            state: absent
+        kernel_settings_sysfs:
+          - name: /sys/kernel/debug/x86/nmi_longest_ns
+            value: 2000000
+        kernel_settings_bootloader_cmdline:
+          - name: spectre_v2
+            value: "on"
+          - name: quiet
+            state: absent
+
+    - name: reboot the machine - see if settings persist after reboot
+      reboot:
+        test_command: tuned-adm active
+
+    - name: check after reboot for ns_longest
+      shell: 'grep 2000000 /sys/kernel/debug/x86/nmi_longest_ns'
```

Comment:
`command: grep 2000000 /sys/kernel/debug/x86/nmi_longest_ns`
We avoid using `shell` if possible - and you don't need the single quotes around the string

Reply:
Hello. Thanks. So what I should use instead?

Reply:
Also: Removed ' chars in another commit.

Reply:
> Hello. Thanks. So what I should use instead?

Use the `command` module instead of `shell`:
```
command: grep 2000000 /sys/kernel/debug/x86/nmi_longest_ns
```
But see my other comment that you can use `tuned-adm verify -i` to verify all settings

Reply:
Thanks. Changed to command everywhere.
It's much easier to handle test results when I see precisely where test failed.